plugin/evm/vm_test.go

```diff
@@ -4092,3 +4104,282 @@ func TestSkipChainConfigCheckCompatible(t *testing.T) {
+					},
+				}
+			},
+			sender2: func(vm1, vm2 *VM) *engCommon.SenderTest {
+				return &engCommon.SenderTest{}
+			},
+		},
+		{
+			name: "pull gossip",
+			sender1: func(vm1, vm2 *VM) *engCommon.SenderTest {
+				return &engCommon.SenderTest{
+					SendAppResponseF: func(ctx context.Context, _ ids.NodeID, requestID uint32, bytes []byte) error {
+						return vm2.AppResponse(ctx, vm1.ctx.NodeID, requestID, bytes)
+					},
+				}
+			},
+			sender2: func(vm1, vm2 *VM) *engCommon.SenderTest {
+				return &engCommon.SenderTest{
+					SendAppRequestF: func(ctx context.Context, _ set.Set[ids.NodeID], requestID uint32, bytes []byte) error {
+						go vm1.AppRequest(ctx, vm2.ctx.NodeID, requestID, time.Time{}, bytes)
+						return nil
+					},
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			vm1 := &VM{}
+			vm1SnowCtx := snow.DefaultContextTest()
+
+			vm2 := &VM{}
+			vm2SnowCtx := snow.DefaultContextTest()
+
+			keyFactory := secp256k1.Factory{}
+			pk, err := keyFactory.NewPrivateKey()
+			require.NoError(err)
+			address := GetEthAddress(pk)
+			genesis := newPrefundedGenesis(params.Ether, address)
+			genesisBytes, err := json.Marshal(genesis)
+			require.NoError(err)
+
+			validatorState := &validators.TestState{
+				GetCurrentHeightF: func(ctx context.Context) (uint64, error) {
+					return 0, nil
+				},
+				GetValidatorSetF: func(context.Context, uint64, ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+					return map[ids.NodeID]*validators.GetValidatorOutput{
+						vm1SnowCtx.NodeID: nil,
+						vm2SnowCtx.NodeID: nil,
+					}, nil
+				},
+			}
+
+			vm1SnowCtx.ValidatorState = validatorState
+			vm2SnowCtx.ValidatorState = validatorState
+
+			require.NoError(vm1.Initialize(
+				context.Background(),
+				vm1SnowCtx,
+				manager.NewMemDB(&version.Semantic{}),
+				genesisBytes,
+				nil,
+				nil,
+				make(chan engCommon.Message),
+				nil,
+				tt.sender1(vm1, vm2),
+			))
+			require.NoError(vm1.SetState(context.Background(), snow.NormalOp))
+
+			require.NoError(vm2.Initialize(
+				context.Background(),
+				vm2SnowCtx,
+				manager.NewMemDB(&version.Semantic{}),
+				genesisBytes,
+				nil,
+				nil,
+				make(chan engCommon.Message),
+				nil,
+				tt.sender2(vm1, vm2),
+			))
+			require.NoError(vm2.SetState(context.Background(), snow.NormalOp))
+
+			tx := types.NewTransaction(0, address, big.NewInt(10), 100_000, big.NewInt(params.LaunchMinGasPrice), nil)
+			signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm1.chainID), pk.ToECDSA())
+			require.NoError(err)
+
+			require.False(vm1.txPool.Has(signedTx.Hash()))
+			require.False(vm2.txPool.Has(signedTx.Hash()))
+
+			require.NoError(vm1.txPool.AddRemote(signedTx))
+			require.True(vm1.txPool.Has(signedTx.Hash()))
+
+			update := make(chan core.NewTxsEvent)
+			vm2.txPool.SubscribeNewTxsEvent(update)
+			<-update
+
+			require.True(vm2.txPool.Has(signedTx.Hash()))
```

Comment:
❤️ this is great

Reply:
Not necessary for this PR, but it would be great to factor out this structure from these two tests so that we can easily re-use this to test vm2vm networking behaviors.